torchci/rockset/pytorch_dev_infra_kpis/__sql/time_to_signal.sql

```diff
@@ -1,68 +1,39 @@
-    from commons.workflow_run w
-    where
-        w.conclusion in ('failure', 'startup_failure', 'cancelled')
-        and w.repository.full_name = 'pytorch/pytorch'
-),
-successful_commits as (
-  select 
-      w.head_commit.id as sha,
-      count(*) as cnt,
-      MIN(w.created_at) as created_at
-  from 
-      commons.workflow_run w left outer join failed_workflows f on w.head_commit.id = f.sha
-  where 
-      f.sha is null
-      and PARSE_TIMESTAMP_ISO8601(w.created_at) > PARSE_TIMESTAMP_ISO8601(:startTime)
-      AND w.repository.full_name = 'pytorch/pytorch'
-      and w.conclusion = 'success'
-      and w.run_attempt = 1 
-  group by sha
-),
-completed_workflows as (
-    select
-        w.created_at as created_at,
-        w.id,
-        w.head_sha
-    from
-        commons.workflow_run w
-    where
-        status = 'completed'
-        and w.repository.full_name = 'pytorch/pytorch'
-        and PARSE_TIMESTAMP_ISO8601(w.created_at) > PARSE_TIMESTAMP_ISO8601(:startTime)
-        and w.run_attempt = 1 
-        and w.head_sha in (select sha from successful_commits)
-),
-tts_by_sha as (
-    select
-        c.head_sha,
-        PARSE_TIMESTAMP_ISO8601(c.created_at) as created_at,
-        max(PARSE_TIMESTAMP_ISO8601(job.completed_at)) as completed_at,
-        count(*) as job_cnt
-    from
-        completed_workflows c HINT(access_path = column_scan)
-        inner join commons.workflow_job job on c.id = job.run_id
-    where
-        name like case
-            when :buildOrAll = 'build' then 'build'
-            else '%'
-        end
-    group by
-        head_sha,
-        created_at
-)
+with
+    tts as (
+        SELECT
+            MAX(
+                DATE_DIFF(
+                    'second',
+                    PARSE_TIMESTAMP_ISO8601(w.created_at),
+                    PARSE_TIMESTAMP_ISO8601(w.updated_at)
+                )
+            ) as duration_sec,
+            w.head_sha,
+            ARBITRARY(IF(w.head_branch = 'main', 'main', 'not main')) as branch,
+            MIN(PARSE_TIMESTAMP_ISO8601(w.created_at)) as created_at
+        FROM
+            commons.workflow_run w
+        WHERE
+            ARRAY_CONTAINS(['pull', 'trunk'], LOWER(w.name))
+            AND PARSE_TIMESTAMP_ISO8601(w.created_at) >= PARSE_DATETIME_ISO8601(:startTime)
+            AND w.head_repository.full_name = 'pytorch/pytorch'
+        group by
+            w.head_sha
+        having
+            bool_and(
+                w.conclusion = 'success'
+                and w.run_attempt = 1
+            )
+    )
 select
-    CAST(DATE_TRUNC('WEEK', created_at) as string) AS week_bucket,
-    avg(
-        DATE_DIFF('minute', created_at, completed_at) / 60.0
-    ) as avg_tts,
+    CAST(DATE_TRUNC('week', t.created_at) as string) AS week_bucket,
+    avg(t.duration_sec / 3600.0) as avg_tts,
+    t.branch
 from
-    tts_by_sha
-where
-    job_cnt > 35
+    tts t
 group by
-    week_bucket
+    week_bucket,
+    t.branch
```

Comment:
Grouping by branch means that every pull request will have the TTS of all it's instances averaged together.

This...is debatably be a good thing.  If one PR gets a single push and CI run, while the second PR is pushed to 20 times, should the 2nd PR's results be weighed 20 times more?  

This implementation is effectively: One PR, one vote.

Reply:
Is this about the old query or the new query?  the t.branch in the new query is `ARBITRARY(IF(w.head_branch = 'main', 'main', 'not main')` so I don't think its one vote per PR

Reply:
Ah, I'd missed that the branch value was a binary.  In that case, do you know what change caused the TTS bump in Jan 2024 to be smoothened out?

Reply:
I believe its from rocm.  The new query only looks at pull and trunk

Here's a graph if you just include rocm
![image](https://github.com/pytorch/test-infra/assets/44682903/0b5d0827-ce6c-4d36-8a96-c490bfaf75b5)